### PR TITLE
chore: updating the poetry version to fix the push issue

### PR DESCRIPTION
**Pull request by cristian-rincon**
This change includes the update on the pyproject.toml file. As the previous release failed due to the package version, as it was the same one that was already present in the Pypi. 

Please refer to this traceback to see the whole error: https://github.com/adriangb/scikeras/actions/runs/6383670304

### CODE REVIEW

**Comment by adriangb** [comment]
So I think the problem here is that removing the pin on grpcio allowed poetry to resolve a newer tensorflow version (see https://github.com/adriangb/scikeras/actions/runs/6424229105/job/17444399467?pr=308#step:6:164, not sure what it was before) and TensorFlow minor updates usually (and unfortunately) have breaking changes. So we'll need the PR to bump the minium TensorFlow version to 2.13 or so and fix any issues that causes. Would you be willing to work on that @cristian-rincon ?

**Comment by cristian-rincon** [comment]
Deprecating tf 2.11 solved several pipelines. I will take a look to see if the current code supports 2.13

**Comment by cristian-rincon** [comment]
Hi @adriangb Upgrading to TensorFlow 13 right now will require several refactors. I cannot afford those refactors now. However, after conducting several tests with different constraints, I have found a solution that works and can fix the pipelines. I suggest we release this version with the new constraints for now. We can then address the upgrade in future releases. What do you think?

File: pyproject.toml [hunk 1]
```diff
@@ -27,15 +27,16 @@ license = "MIT"
...
+version = "0.11.1"
 
 [tool.poetry.dependencies]
 importlib-metadata = {version = ">=3", python = "<3.8"}
 python = ">=3.8.0,<3.12.0"
 scikit-learn = ">=1.0.0"
 packaging = ">=0.21"
-tensorflow = {version = ">=2.11.0", optional = true}
-tensorflow-cpu = {version = ">=2.11.0", optional = true}
+tensorflow = {version = "<=12.0.0", optional = true}
```
Comment (adriangb) [inline pyproject.toml:37]: Is this version a typo? Did you mean `2.12.0`?

Comment (cristian-rincon) [reply pyproject.toml]: You're right! i will apply the refactor now

Comment (cristian-rincon) [reply pyproject.toml]: Done! ✅ @adriangb

File: pyproject.toml [hunk 2]
```diff
@@ -27,15 +27,16 @@ license = "MIT"
...
 
 [tool.poetry.dependencies]
 importlib-metadata = {version = ">=3", python = "<3.8"}
 python = ">=3.8.0,<3.12.0"
 scikit-learn = ">=1.0.0"
 packaging = ">=0.21"
-tensorflow = {version = ">=2.11.0", optional = true}
-tensorflow-cpu = {version = ">=2.11.0", optional = true}
+tensorflow = {version = "<=2.12.1", optional = true}
+tensorflow-cpu = {version = "<=2.12.1", optional = true}
```
Comment (adriangb) [inline pyproject.toml:38]: Why `2.12.1`? Can we be compatible with any `2.12.X` version at least?

Comment (cristian-rincon) [reply pyproject.toml]: Done ✅

Comment (cristian-rincon) [reply pyproject.toml]: @adriangb

File: pyproject.toml [hunk 3]
```diff
@@ -27,15 +27,16 @@ license = "MIT"
 name = "scikeras"
 readme = "README.md"
 repository = "https://github.com/adriangb/scikeras"
-version = "0.11.0"
+version = "0.11.1"
```
Comment (adriangb) [inline pyproject.toml:30]: Lets make this a minor version bump.

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -111,8 +111,8 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        tf-version: [2.11.0]
-        python-version: ["3.7", "3.9"]
+        tf-version: [2.12.0]
+        python-version: ["3.8", "3.9"]
```
Comment (adriangb) [inline .github/workflows/tests.yaml:115]: Does 2.12 not support 3.7?

Comment (cristian-rincon) [reply .github/workflows/tests.yaml:115]: Hi, unfortunately no.

**Review by adriangb — APPROVED** [review]

**Pull request merged**